plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -381,12 +383,22 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             throw;
          };
 
+         // Notes:
+         // - as soon as block is accepted via `push_block` a signal is emitted.
+         // - may be able to just overload `push_block` to _verify_ the block if blockvault is currently active; will have to just pass noop lambda:
+         //   - this is no longer planned.
          try {
             chain.push_block( bsf, [this]( const branch_type& forked_branch ) {
                _unapplied_transactions.add_forked( forked_branch );
             }, [this]( const transaction_id_type& id ) {
                return _unapplied_transactions.get_trx( id );
             } );
+            if ( blockvault_plug ) {
+               uint32_t lib{chain.head_block_state()->dpos_irreversible_blocknum};
```

Comment:
`head_block_state` is not necessarily the block you just pushed. Could return the `block_state_ptr` from `push_block`, or eval the future before calling `push_block`, or get it from the `accepted_block_header` signal.

Reply:
Changed the code to use the `block_state_ptr` returned from evaluating the future `bsf`.